src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/CompilerServices/Operators.vb

```diff
@@ -4610,27 +4591,18 @@ Namespace Microsoft.VisualBasic.CompilerServices
         End Function
 
         Private Shared Function ModInt16(ByVal left As Int16, ByVal right As Int16) As Object
-            Dim result As Integer = CInt(left) Mod CInt(right)
-
-            If result < Int16.MinValue OrElse result > Int16.MaxValue Then
-                Return result
-            Else
-                Return CShort(result)
-            End If
+            Return left Mod right
         End Function
 
         Private Shared Function ModUInt16(ByVal left As UInt16, ByVal right As UInt16) As Object
             Return left Mod right
         End Function
 
         Private Shared Function ModInt32(ByVal left As Integer, ByVal right As Integer) As Object
-            'Do operation with Int64 to avoid OverflowException with Int32.MinValue and -1
-            Dim result As Long = CLng(left) Mod CLng(right)
-
-            If result < Int32.MinValue OrElse result > Int32.MaxValue Then
-                Return result
+            If left = Integer.MinValue AndAlso right = -1 Then
```

Comment:
This can't overflow, and the degenerate case, if left == Int16.MinValue and right == -1 does not overflow and returns 0

But I have special cased the degenerate case to match `ModInt64`:
```vb
         Private Shared Function ModInt64(ByVal left As Int64, ByVal right As Int64) As Object
 
             If left = Int64.MinValue AndAlso right = -1 Then
                 Return 0L
             Else
                 Return left Mod right
             End If

```